dspace-server-webapp/src/test/java/org/dspace/app/rest/RegistrationRestControllerIT.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.sql.SQLException;
+import java.util.Iterator;
+import java.util.List;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.dspace.app.rest.model.RegistrationRest;
+import org.dspace.app.rest.test.AbstractControllerIntegrationTest;
+import org.dspace.eperson.RegistrationData;
+import org.dspace.eperson.dao.RegistrationDataDAO;
+import org.dspace.services.ConfigurationService;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class RegistrationRestControllerIT extends AbstractControllerIntegrationTest {
+
+    @Autowired
+    private RegistrationDataDAO registrationDataDAO;
+
+    @Autowired
+    private ConfigurationService configurationService;
+
+
+    @Before
+    public void setup() throws SQLException {
+        CollectionUtils.emptyIfNull(registrationDataDAO.findAll(context, RegistrationData.class)).stream()
+                       .forEach(registrationData -> {
+                           try {
+                               registrationDataDAO.delete(context, registrationData);
+                           } catch (SQLException e) {
+                               throw new RuntimeException(e);
+                           }
+                       });
+    }
```

Comment:
Since you are now cleaning up the created registration data in each of the below tests (and in other tests in this PR), this entire `setup()` should be unnecessary & likely can be removed.  Each test should cleanup after itself instead of doing "just in case" cleanup before running each test.

Reply:
This has been removed now